File: TwoPointer/647.palindromic-substrings.py
# @leet start
class Solution:
    def countSubstrings(self, s: str) -> int:
        ret = 0

        def search(l, r):
            while l >= 0 and r < len(s):
                if s[l] != s[r]:  # NOTE: need break
                    break
                if s[l] == s[r]:
                    nonlocal ret
                    ret += 1
                l -= 1
                r += 1

        for i in range(len(s)):
            search(i, i)
            search(i, i + 1)
        return ret


# @leet end

File: TwoPointer/647.palindromic-substrings.ts
// @leet start
function countSubstrings(s: string): number {
  let ret = 0;
  function check(l: number, r: number) {
    while (l >= 0 && r < s.length && s[l] === s[r]) {
      if (s[l] == s[r]) {
        ret++;
      }
      l--;
      r++;
    }
  }
  for (let i = 0; i < s.length; i++) {
    check(i, i);
    check(i, i + 1);
  }

  return ret;
}
// @leet end

